architecture/transaction-layer.md

```diff
@@ -0,0 +1,204 @@
+
+### client.Txn and TxnCoordSender
+
+As we mentioned in the SQL layer's architectural overview, CockroachDB converts all SQL statements into key-value (KV) operations, which is how data is ultimately stored and accessed. 
+
+All of the KV operations generated from the SQL layer's use `client.Txn`, which is the transactional interface for the CockroachDB KV layer––but, as we discussed above, all statements are treated as transactions, so all statements use this interface.
+
+However, `client.Txn` is actually just a wrapper around `TxnCoordSender`, which plays a crucial role in our code base by:
+
+- Dealing with transactions' state. After a transaction is started, `TxnCoordSender` starts asynchronously sending heartbeat messages to that transaction's Transaction Record, which signal that it should be kept alive. If the `TxnCoordSender`'s heartbeating stops, the Transaction Record is moved to `ABORTED` status.
+- Tracking each written key or key range over the course of the transaction. 
+- Clearing the accumulated Write Intent for the transaction when it's committed or aborted. All requests being performed as part of a transaction have to go through the same `TxnCoordSender` to account for all of its Write Intents, which optimizes the cleanup process.
+
+After setting up this bookkeeping, the request is passed to the `DistSender` in the Distribution Layer.
+
+### Transaction Records
+
+When a transaction starts, `TxnCoordSender` writes a Transaction Record to the range containing the first key modified in the transaction. As mentioned above, the Transaction Record provides the system with a source of truth about the status of a transaction.
+
+The Transaction Record expresses one of the following dispositions of a transaction:
+
+- `PENDING`: The initial status of all values, indicating that the transaction the Write Intent belongs to is in progress. Note that retried transactions remain in this status because we continue to increment the transaction's priority in an attempt to cause transaction's likelihood of succeeding to increase the longer they remain in the system.
+- `COMMITTED`: Once a transaction has completed, this status indicates that the value can be read. Whenever this status is encountered, the transaction status attribute is removed, converting the value from a Write Intent to a standard MVCC value.
+- `ABORTED`: If a transaction fails or is aborted by the client, it's moved into this state.
+
+This record remains on the coordinating node until all of the Write Intents it's responsible for are converted to MVCC values.
+
+The Transaction Record for committed transactions remain until all Write Intents are converted to MVCC values. For aborted transactions, the transaction record can be deleted at any time, which also means that CockroachDB treats missing transaction records as if they belong to aborted transactions.
```

Comment:
Count is inconsistent. I guess it's best to make singular, since the rest of this section goes that route.